server/src/main/java/org/opensearch/search/approximate/ApproximateConstantScoreWeight.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.search.approximate;
+
+import org.apache.lucene.index.PointValues;
+import org.apache.lucene.search.ConstantScoreWeight;
+import org.apache.lucene.search.Query;
+import org.apache.lucene.util.DocIdSetBuilder;
+
+import java.io.IOException;
+
+public abstract class ApproximateConstantScoreWeight extends ConstantScoreWeight {
```

Comment:
isn't it only applicable to `Point` based queries?  calling it `ApproximateConstantScoreWeight ` is giving an impression of its general applicability.

Reply:
I changed this to `ApproximatePointRangeQueryWeight`, which implicitly extends `ConstantScoreWeight`.